lib/matplotlib/backends/backend_pgf.py

```diff
@@ -29,7 +29,9 @@
 _log = logging.getLogger(__name__)
 
 
-_DOCUMENTCLASS = r"\documentclass{article}"
+_DOCUMENTCLASS = (
+    rf"\documentclass{{{mpl.rcParams.get('pgf.documentclass', 'article')}}}"
+)
```

Comment:
I think the reason that this is a variable is that it should be possible to override the document class. (Although in an undocumented way.)

Since it is now possible to do that in a more regular way, it would make sense to rewrite the code to not use the variable any more. It is used in three locations though, so maybe one should simply try to factor out the pre-amble code in some way. Which may lead to that your current solution can stay...